pkg/controller/atlasproject/ipaccess_list.go

```diff
@@ -0,0 +1,129 @@
+		onlyOneSpecified := isNotEmpty(list.AwsSecurityGroup) != isNotEmpty(list.CIDRBlock) != isNotEmpty(list.IPAddress)
+		if !onlyOneSpecified {
+			return errors.New("only one of the 'awsSecurityGroup', 'cidrBlock' or 'ipAddress' is required be specified")
+		}
+	}
+	return nil
+}
+
+func createOrDeleteInAtlas(client *mongodbatlas.Client, projectID string, operatorIPAccessLists []mdbv1.ProjectIPAccessList, log *zap.SugaredLogger) workflow.Result {
+	atlasAccess, _, err := client.ProjectIPAccessList.List(context.Background(), projectID, &mongodbatlas.ListOptions{})
+	if err != nil {
+		return workflow.Terminate(workflow.ProjectIPNotCreatedInAtlas, err.Error())
+	}
+	// Making a new slice with synonyms as Atlas IP Access list to enable usage of 'Identifiable'
+	atlasAccessLists := make([]atlasProjectIPAccessList, len(atlasAccess.Results))
+	for i, r := range atlasAccess.Results {
+		atlasAccessLists[i] = atlasProjectIPAccessList(r)
+	}
+
+	difference := set.Difference(atlasAccessLists, operatorIPAccessLists)
+
+	if err := deleteIPAccessFromAtlas(client, projectID, difference, log); err != nil {
+		return workflow.Terminate(workflow.ProjectIPNotCreatedInAtlas, err.Error())
+	}
+
+	if result := createIPAccessListsInAtlas(client, projectID, operatorIPAccessLists); !result.IsOk() {
+		return result
+	}
+	return workflow.OK()
+}
+
+func createIPAccessListsInAtlas(client *mongodbatlas.Client, projectID string, ipAccessLists []mdbv1.ProjectIPAccessList) workflow.Result {
+	operatorAccessLists := make([]*mongodbatlas.ProjectIPAccessList, len(ipAccessLists))
+	for i, list := range ipAccessLists {
+		atlasFormat, err := list.ToAtlas()
+		if err != nil {
+			return workflow.Terminate(workflow.Internal, err.Error())
+		}
+		operatorAccessLists[i] = atlasFormat
+	}
+
+	if _, _, err := client.ProjectIPAccessList.Create(context.Background(), projectID, operatorAccessLists); err != nil {
+		return workflow.Terminate(workflow.ProjectIPNotCreatedInAtlas, err.Error())
+	}
+	return workflow.OK()
+}
+
+func deleteIPAccessFromAtlas(client *mongodbatlas.Client, projectID string, listsToRemove []set.Identifiable, log *zap.SugaredLogger) error {
+	for _, l := range listsToRemove {
+		if _, err := client.ProjectIPAccessList.Delete(context.Background(), projectID, l.Identifier().(string)); err != nil {
+			return err
+		}
+		log.Debugw("Removed IPAccessList from Atlas as it's not specified in current AtlasProject", "id", l.Identifier())
+	}
+	return nil
+}
+
+func filterActiveIPAccessLists(project *mdbv1.AtlasProject) ([]mdbv1.ProjectIPAccessList, []mdbv1.ProjectIPAccessList) {
+	active := make([]mdbv1.ProjectIPAccessList, 0)
+	expired := make([]mdbv1.ProjectIPAccessList, 0)
+	for _, list := range project.Spec.ProjectIPAccessList {
+		if list.DeleteAfterDate != "" {
+			// We are ignoring the error as it will never happen due to validation check before
+			iso8601, _ := timeutil.ParseISO8601(list.DeleteAfterDate)
```

Comment:
This will error out if time format is different, which can happen in theory since ISO 8601 can range in precision and the presence of TZ info (see below)

Reply:
yes, you are right - this is fixed in the next PR!